internal/commands/test_test.go

```diff
@@ -167,3 +170,50 @@ ENTRYPOINT ["java","-cp","app:app/lib/*","hello.Application"]`
 		t.Errorf("Dockerfile test failure. Got %v successes, expected %v", actualSuccesses, expectedSuccesses)
 	}
 }
+
+func TestGetFilesFromDirectory(t *testing.T) {
+	os.Mkdir("test/", 0700)
+	defer os.RemoveAll("test/")
+
+	createDummyFile := func(name string) {
+		d := []byte("")
+		check(ioutil.WriteFile(name, d, 0644))
+	}
+	err := os.MkdirAll("test/parent/child", 0755)
+	check(err)
+
+	createDummyFile("test/file1.tf")
+	createDummyFile("test/file2.tf")
+	createDummyFile("test/parent/file1.tf")
+	createDummyFile("test/parent/file1.yaml")
+	createDummyFile("test/parent/child/test.tf")
+
+	tests := []struct {
+		regex string
+		exp   []string
+	}{
+		{".*.yaml", []string{"test/file1.tf", "test/file2.tf", "test/parent/child/test.tf", "test/parent/file1.tf"}},
+		{".*.tf", []string{"test/parent/file1.yaml"}},
+		{"child/", []string{"test/file1.tf", "test/file2.tf", "test/parent/file1.tf", "test/parent/file1.yaml"}},
+		{"parent/", []string{"test/file1.tf", "test/file2.tf"}},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.regex, func(t *testing.T) {
+			result, err := getFilesFromDirectory("test/", tt.regex)
+			if err != nil {
+				t.Errorf("getFilesFromDirectory returns err, expected nil")
+			}
+
+			if !reflect.DeepEqual(tt.exp, result) {
+				t.Errorf("expected: %v, got: %v", tt.exp, result)
+			}
+		})
+	}
+}
+
+func check(e error) {
```

Comment:
nit: Rather than introducing a new check function that throws a panic, it feels a bit more idiomatic to catch the error and use t.Fatalf within the test.

Reply:
agreed on that as well, `check` is now removed